### PR TITLE
HBASE-22590 Remove the deprecated methods in Table interface

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/client/ThriftTable.java [hunk 1]
```diff
@@ -410,8 +410,7 @@ public boolean thenMutate(RowMutations mutation) throws IOException {
   }
 
 
-  @Override
-  public boolean checkAndMutate(byte[] row, byte[] family, byte[] qualifier, CompareOperator op,
+ private boolean checkAndMutate(byte[] row, byte[] family, byte[] qualifier, CompareOperator op,
```
Comment (HorizonNet) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/client/ThriftTable.java]: Why does this one get private?

Comment (Apache9) [reply hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/client/ThriftTable.java]: It is called by other methods in this class so we can not just remove it...

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 70 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 20 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 258 | master passed |
| +1 | compile | 215 | master passed |
| +1 | checkstyle | 203 | master passed |
| +1 | shadedjars | 277 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 475 | master passed |
| +1 | javadoc | 148 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 250 | the patch passed |
| +1 | compile | 212 | the patch passed |
| +1 | javac | 212 | the patch passed |
| -1 | checkstyle | 31 | hbase-client: The patch generated 13 new + 56 unchanged - 13 fixed = 69 total (was 69) |
| +1 | checkstyle | 75 | hbase-server: The patch generated 0 new + 107 unchanged - 25 fixed = 107 total (was 132) |
| +1 | checkstyle | 18 | The patch passed checkstyle in hbase-mapreduce |
| -1 | checkstyle | 34 | hbase-thrift: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | checkstyle | 13 | The patch passed checkstyle in hbase-backup |
| +1 | checkstyle | 15 | The patch passed checkstyle in hbase-it |
| +1 | checkstyle | 14 | hbase-rest: The patch generated 0 new + 10 unchanged - 122 fixed = 10 total (was 132) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 274 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 781 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 496 | the patch passed |
| +1 | javadoc | 149 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 190 | hbase-client in the patch passed. |
| +1 | unit | 7692 | hbase-server in the patch passed. |
| +1 | unit | 965 | hbase-mapreduce in the patch passed. |
| +1 | unit | 195 | hbase-thrift in the patch passed. |
| +1 | unit | 899 | hbase-backup in the patch passed. |
| +1 | unit | 66 | hbase-it in the patch passed. |
| -1 | unit | 291 | hbase-rest in the patch failed. |
| +1 | asflicense | 211 | The patch does not generate ASF License warnings. |
| | | 14984 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.rest.client.TestRemoteTable |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/309 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 968857414c0f 4.4.0-131-generic #157~14.04.1-Ubuntu SMP Fri Jul 13 08:53:17 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 1712d2b568 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/artifact/out/diff-checkstyle-hbase-client.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/artifact/out/diff-checkstyle-hbase-thrift.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/artifact/out/patch-unit-hbase-rest.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/testReport/ |
| Max. process+thread count | 5430 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-mapreduce hbase-thrift hbase-backup hbase-it hbase-rest U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 96 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 21 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 39 | Maven dependency ordering for branch |
| +1 | mvninstall | 249 | master passed |
| +1 | compile | 210 | master passed |
| +1 | checkstyle | 196 | master passed |
| +1 | shadedjars | 259 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 460 | master passed |
| +1 | javadoc | 142 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 240 | the patch passed |
| +1 | compile | 214 | the patch passed |
| +1 | javac | 214 | the patch passed |
| +1 | checkstyle | 32 | hbase-client: The patch generated 0 new + 56 unchanged - 13 fixed = 56 total (was 69) |
| +1 | checkstyle | 71 | hbase-server: The patch generated 0 new + 108 unchanged - 26 fixed = 108 total (was 134) |
| +1 | checkstyle | 18 | The patch passed checkstyle in hbase-mapreduce |
| +1 | checkstyle | 31 | The patch passed checkstyle in hbase-thrift |
| +1 | checkstyle | 16 | The patch passed checkstyle in hbase-backup |
| +1 | checkstyle | 14 | The patch passed checkstyle in hbase-it |
| +1 | checkstyle | 16 | hbase-rest: The patch generated 0 new + 9 unchanged - 123 fixed = 9 total (was 132) |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedjars | 262 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 792 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 457 | the patch passed |
| +1 | javadoc | 139 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 204 | hbase-client in the patch passed. |
| +1 | unit | 7877 | hbase-server in the patch passed. |
| +1 | unit | 919 | hbase-mapreduce in the patch passed. |
| +1 | unit | 177 | hbase-thrift in the patch passed. |
| +1 | unit | 807 | hbase-backup in the patch passed. |
| +1 | unit | 66 | hbase-it in the patch passed. |
| +1 | unit | 287 | hbase-rest in the patch passed. |
| +1 | asflicense | 255 | The patch does not generate ASF License warnings. |
| | | 14951 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/309 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 1b23bc68630a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 1712d2b568 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/3/artifact/out/whitespace-eol.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/3/testReport/ |
| Max. process+thread count | 5793 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-mapreduce hbase-thrift hbase-backup hbase-it hbase-rest U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Ah, whitespace...

Let me check...

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 21 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for branch |
| +1 | mvninstall | 237 | master passed |
| +1 | compile | 212 | master passed |
| +1 | checkstyle | 185 | master passed |
| +1 | shadedjars | 271 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 465 | master passed |
| +1 | javadoc | 156 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 238 | the patch passed |
| +1 | compile | 213 | the patch passed |
| +1 | javac | 213 | the patch passed |
| +1 | checkstyle | 31 | hbase-client: The patch generated 0 new + 56 unchanged - 13 fixed = 56 total (was 69) |
| +1 | checkstyle | 69 | hbase-server: The patch generated 0 new + 108 unchanged - 26 fixed = 108 total (was 134) |
| +1 | checkstyle | 18 | The patch passed checkstyle in hbase-mapreduce |
| +1 | checkstyle | 33 | The patch passed checkstyle in hbase-thrift |
| +1 | checkstyle | 15 | The patch passed checkstyle in hbase-backup |
| +1 | checkstyle | 13 | The patch passed checkstyle in hbase-it |
| +1 | checkstyle | 15 | hbase-rest: The patch generated 0 new + 9 unchanged - 123 fixed = 9 total (was 132) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 252 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 719 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 503 | the patch passed |
| +1 | javadoc | 144 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 134 | hbase-client in the patch passed. |
| +1 | unit | 8854 | hbase-server in the patch passed. |
| +1 | unit | 1017 | hbase-mapreduce in the patch passed. |
| +1 | unit | 178 | hbase-thrift in the patch passed. |
| +1 | unit | 809 | hbase-backup in the patch passed. |
| +1 | unit | 66 | hbase-it in the patch passed. |
| +1 | unit | 297 | hbase-rest in the patch passed. |
| +1 | asflicense | 203 | The patch does not generate ASF License warnings. |
| | | 15818 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/309 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 774e366d2318 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / ed30909d27 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/5/testReport/ |
| Max. process+thread count | 5768 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-mapreduce hbase-thrift hbase-backup hbase-it hbase-rest U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-309/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**